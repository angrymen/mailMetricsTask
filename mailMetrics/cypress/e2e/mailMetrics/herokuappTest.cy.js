/// <reference types = "cypress" />
/// <reference types = "cypress-xpath" />
/// <reference types = "cypress-plugin-snapshots" />
const { deleteDownloadsFolderBeforeEach } = require('cypress-delete-downloads-folder');


describe("visual regresion", () => {

  deleteDownloadsFolderBeforeEach()

  it.only("visual regresion - test herokuApp", () => {
    //6. Then go to http://the-internet.herokuapp.com/dynamic_content?with_content=static, click 
    //on “click here”, wait until content is refreshed and count number of characters in the last 
    //section which gets refreshed

    cy.visit(
      Cypress.env("herokuApp")
    );
    cy.document().then((image) => {
      cy.wrap(image)
        .toMatchSnapshot()
        .as("zd1")
        .then(() => {
          cy.xpath("//p//a")
            .click()
            .then(() => {
              cy.document().then((imageAfterRefresh) => {
                cy.wrap(imageAfterRefresh)
                  .toMatchSnapshot()
                  .as("zd2");
              });
            });
        });
    });
    cy.get("@zd1")
      .should("not.equal", "@zd2")
      .then(() => {
        cy.log("Test pass");
      });
      cy.xpath('//*[@class="large-10 columns"]').eq(2).invoke("text").then((getText) => {
        cy.log("The length of a string: " + getText.length)
      })
      
  });

// second methods
  it('visual regresion - test second method herokuApp', () => {
    cy.xpath('//*[@class="large-2 columns"]').eq(2).then(image =>{
        cy.wrap(image).toMatchSnapshot().as('zd1')
        .then(()=>{
            cy.xpath('//p//a').click().then(()=>{
                cy.xpath('//*[@class="large-2 columns"]').eq(2).then((imageAfterRefresh)=>{
                    cy.wrap(imageAfterRefresh).toMatchSnapshot().as('zd2')
                })
            })

        })
    })
     cy.get('@zd1').should('not.equal','@zd2').then (() => {
        cy.log("Test pass")
     })
     cy.xpath('//*[@class="large-10 columns"]').eq(2).invoke("text").then((getText) => {
        cy.log("The length of a string: " + getText.length)
      })
  });
});