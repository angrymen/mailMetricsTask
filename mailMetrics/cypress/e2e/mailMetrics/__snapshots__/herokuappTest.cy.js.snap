exports[`visual regresion > visual regresion - test herokuApp #0`] =
{
  "jQuery111304246473638588888": 1,
  "location": {
    "ancestorOrigins": {
      "0": "http://the-internet.herokuapp.com"
    },
    "hash": "",
    "host": "the-internet.herokuapp.com",
    "hostname": "the-internet.herokuapp.com",
    "href": "http://the-internet.herokuapp.com/dynamic_content?with_content=static",
    "origin": "http://the-internet.herokuapp.com",
    "pathname": "/dynamic_content",
    "port": "",
    "protocol": "http:",
    "search": "?with_content=static"
  },
  "referrer": ""
};

exports[`visual regresion > visual regresion - test herokuApp #1`] =
{
  "jQuery11130411117664926381": 1,
  "location": {
    "ancestorOrigins": {
      "0": "http://the-internet.herokuapp.com"
    },
    "hash": "",
    "host": "the-internet.herokuapp.com",
    "hostname": "the-internet.herokuapp.com",
    "href": "http://the-internet.herokuapp.com/dynamic_content?with_content=static",
    "origin": "http://the-internet.herokuapp.com",
    "pathname": "/dynamic_content",
    "port": "",
    "protocol": "http:",
    "search": "?with_content=static"
  }
};
